src/jitlog/reader.lua

```diff
@@ -24,6 +23,77 @@ function base_actions:stringmarker(buff)
   }
   self.markers[#self.markers + 1] = marker
   self:log_msg("stringmarker", "StringMarker: %s %s", label, time)
+  return marker
+end
+
```

Comment:
the whole src/jitlog directory has grown pretty complex.  i don't know what parts are infrastructure (i guess the generator files), what are a usable toolkit (messages.lua, reader lua ?) and what are minimal example code (as i thought reader.lua was).

Reply:
Just reader.lua and the generated reader_def.lua are needed to read the log files but the generator.lua + messages.lua allows reflection over the message fields if needed. That return value is passed on to any mixins that registered to also handle the message